set.mm

```diff
@@ -79592,21 +79592,544 @@ currently used conventions for such cases (see ~ cbvmpox , ~ ovmpox and
     ( wcel cund cfv cuni cpw c0 undefval wne pwne0 a1i eqnetrd ) ABCZADEAFZGZHA
     BIPHJNOKLM $.
 
-
 $(
 =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
   Well-founded recursion
 =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 $)
 
+
+  $( Declare new constant symbols. $)
+  $c frecs $.
+
+  $( Declare the syntax for the well-founded recursion generator.  See
+     ~ df-frecs . $)
+  cfrecs $a class frecs ( R , A , F ) $.
+
+  ${
+    $d R f x y $.  $d A f x y $.  $d F f x y $.
+    $( This is the definition for the well-founded recursion generator.
+       Similar to ~ df-wrecs and ~ df-recs , it is a direct definition form of
+       normally recursive relationships.  Unlike the former two definitions, it
+       only requires a founded set-like relationship for its properties, not a
+       well-founded relationship.  When this relationship is also a partial
+       ordering, the proof does not use the axiom of infinity, but it requires
+       the axioms when the order is not partial.  We develop the theorems
+       twice, once with partial ordering and once without.  The second
+       development occurs later in the database, after ~ ax-inf has been
+       introduced.  (Contributed by Scott Fenton, 23-Dec-2021.) $)
+    df-frecs $a |- frecs ( R , A , F ) = U. { f | E. x ( f Fn x /\ ( x C_ A /\
+               A. y e. x Pred ( R , A , y ) C_ x ) /\
+               A. y e. x ( f ` y ) = ( y F ( f |` Pred ( R , A , y ) ) ) ) } $.
+  $}
+
+  ${
+    $d R f x y $.  $d S f x y $.  $d A f x y $.  $d B f x y $.  $d F f x y $.
+    $d G f x y $.
+    $( Equality theorem for the well-founded recursion generator.  (Contributed
+       by Scott Fenton, 23-Dec-2021.) $)
+    frecseq123 $p |- ( ( R = S /\ A = B /\ F = G ) ->
+      frecs ( R , A , F ) = frecs ( S , B , G ) ) $=
+      ( vf vx vy wceq w3a cv wss cpred wral wa cres co wex cab wfn cfrecs simp2
+      cfv cuni sseq2d weq equid predeq123 mp3an3 3adant3 sseq1d ralbidv anbi12d
+      simp3 oveqd reseq2d oveq2d eqeq2d 3anbi23d exbidv abbidv df-frecs 3eqtr4g
+      eqtrd unieqd ) CDJZABJZEFJZKZGLZHLZUAZVLAMZACILZNZVLMZIVLOZPZVOVKUDZVOVKV
+      PQZERZJZIVLOZKZHSZGTZUEVMVLBMZBDVONZVLMZIVLOZPZVTVOVKWIQZFRZJZIVLOZKZHSZG
+      TZUEACEUBBDFUBVJWGWSVJWFWRGVJWEWQHVJVSWLWDWPVMVJVNWHVRWKVJABVLVGVHVIUCUFV
+      JVQWJIVLVJVPWIVLVGVHVPWIJZVIVGVHIIUGWTIUHABCDVOVOUIUJUKZULUMUNVJWCWOIVLVJ
+      WBWNVTVJWBVOWAFRWNVJEFVOWAVGVHVIUOUPVJWAWMVOFVJVPWIVKXAUQURVEUSUMUTVAVBVF
+      HIACGEVCHIBDGFVCVD $.
+  $}
+
+  ${
+    $d R f y z $.  $d A f y z $.  $d F f y z $.  $d x f y z $.
+    nffrecs.1 $e |- F/_ x R $.
+    nffrecs.2 $e |- F/_ x A $.
+    nffrecs.3 $e |- F/_ x F $.
+    $( Bound-variable hypothesis builder for the well-founded recursion
+       generator.  (Contributed by Scott Fenton, 23-Dec-2021.) $)
+    nffrecs $p |- F/_ x frecs ( R , A , F ) $=
```

Comment:
(And that particular theorem is not a very good example, since the need of deduction form would be very rare: we have the choice to use a fresh variable x; for other theorems, there may be a need.)